src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURE_COUNT = 5;
+    private static final int UPPER_INTEGER_BOUND = 10;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        switch (random.nextInt(FIGURE_COUNT)) {
+            case 0:
+                return new Circle(colorSupplier.getRandomColor(), getRandomIntValue());
```

Comment:
lets add private methods to get each random figure - this would make your code easier to read

```suggestion
            case 0:
                return getRandomCircle();
```